File: sample/javaScript/turtleSpiral.mjs
// import { Voxelamming, Turtle } from 'voxelamming';
import { Turtle } from 'voxelamming'; // test
import Voxelamming from './voxelamming.js';  // test

const roomName = "1000";
const voxelamming = new Voxelamming(roomName);

voxelamming.setBoxSize(0.3);
voxelamming.setBuildInterval(0.001);
voxelamming.setCommand('liteRender')
const t = new Turtle(voxelamming);

const colors = [
  [0, 0, 0, 1],
  [1, 0, 0, 1],
  [0, 1, 0, 1],
  [0, 0, 1, 1],
  [1, 1, 0, 1],
  [0, 1, 1, 1],
  [1, 0, 1, 1],
  // [1, 1, 1, 1],
  // [0.5, 0, 0, 1],
  // [0, 0.5, 0, 1],
  // [0, 0, 0.5, 1],
  // [0.5, 0.5, 0, 1],
  // [0, 0.5, 0.5, 1],
  // [0.5, 0, 0.5, 1],
  // [0.5, 0.5, 0.5, 1],
];

for (let i = 0; i < colors.length; i++) {
  const color = colors[i];
  const polarPhi = (i * 180) / colors.length;
  t.reset();
  t.setColor(...color);
  t.setPos(i, 0, 0);
  t.up(4);

  for (let _ = 0; _ < 360; _++) {
    t.forward(3);
    t.left(6);
  }
}

await voxelamming.sendData();
console.log('send data done');

File: sample/python/turtle_spiral.py
# from voxelamming import Voxelamming, Turtle
from voxelamming_local import Voxelamming
from turtle import Turtle

# Voxelammingアプリに表示されている部屋名を指定してください
room_name = "1000"
# Voxelammingクラスのインスタンスを生成します
voxelamming = Voxelamming(room_name)
# ボクセルの設定を行います
voxelamming.set_box_size(0.3)
voxelamming.set_build_interval(0.001)
voxelamming.set_command('liteRender')  # 描画を軽くするためのコマンド

# ボクセルを配置するため、位置と色を設定します
t = Turtle(voxelamming)

# 線の色のリスト
colors = [
    [0, 0, 0, 1],
    [1, 0, 0, 1],
    [0, 1, 0, 1],
    [0, 0, 1, 1],
    [1, 1, 0, 1],
    [0, 1, 1, 1],
    [1, 0, 1, 1]
]

# 色を変えながら螺旋を描画します
for i, color in enumerate(colors):
    t.reset()
    t.set_color(*color)
    t.set_pos(i, 0, 0)
    t.up(4)

    for _ in range(360):
        t.forward(3)
        t.left(6)

# ボクセルデータをアプリに送信します。
voxelamming.send_data("turtle_spiral")
